### PR TITLE
Matching email notifications

### PR DESCRIPTION
**Pull request by androane**
I created a model to keep email addresses to be notified when companies matching occurs.
The DB of contacts(email, first name and last name) can be modified through the API
Also with tests

### CODE REVIEW

File: fcs/mails.py [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
+from flask import current_app as app, render_template
 
+from fcs.models import MailAddress
 
-def send_mail(subject, html):
-    recipients = app.config.get('NOTIFY_EMAILS')
-    sender = 'Eau de Web <%s>' % app.config.get('MAIL_USERNAME')
+
+def send_mail(subject, html, recipients):
+    sender = 'BDR Help Desk'
```
Comment (alexef) [inline fcs/mails.py]: Put it in config also.

File: fcs/templates/mails/match_notification.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  Dear {{ first_name }} {{ last_name }},
+
+  <p>
+    User <b>{{ user }}</b> has matched <b>{{ company_name }}</b> company (registered in the FGAS portal)
+    with the old record <b>{{ oldcompany_name }}</b> (taken from BDR-Registry), based on suggestions made by the system.
+
+    <br>
+
+    You can see the company details here:
+    <a href="https://bdr-test.eionet.europa.eu/fgases_registry/organisation_details?id={{ company_id }}">
```
Comment (alexef) [inline fcs/templates/mails/match_notification.html]: Put the hostname part of the URL in config.

File: fcs/templates/mails/no_match_notification.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  Dear {{ first_name }} {{ last_name }},
+
+  <p>
+    User <b>{{ user }}</b> has added <b>{{ company_name }}</b> company (registered in the FGAS portal).
+    It had not match with any of the old records.
+
+    <br>
+
+    You can see the company details here:
+    <a href="https://bdr-test.eionet.europa.eu/fgases_registry/organisation_details?id={{ company_id }}">
```
Comment (alexef) [inline fcs/templates/mails/no_match_notification.html]: same

File: settings.py.example [hunk 4]
```diff
@@ -24,8 +24,9 @@ MAIL_USE_SSL = True
 MAIL_USERNAME = ''
 MAIL_PASSWORD = ''
 
-NOTIFY_EMAILS = ['mihai.zamfir@eaudeweb.ro']
 
 HTTPS_VERIFY = None # set it to False to disable
 
 AUTO_VERIFY_NEW_COMPANIES = False
+
+SEND_MATCHING_MAILS = True
```
Comment (alexef) [inline settings.py.example]: Put it in app.py to be False by default.

**Comment by alexef** [comment]
Is this one merged yet?

**Comment by androane** [comment]
No, it's not

**Pull request merged**